data/kernels/basic.cl

```diff
@@ -1070,14 +1070,15 @@ lens_distort_bilinear (read_only image2d_t in, write_only image2d_t out, const i
 
   pixel = all(isfinite(pixel.xyz)) ? pixel : (float4)0.0f;
 
+  if(monochrome) pixel.x = pixel.z = pixel.y;
```

Comment:
So in monochrome mode we expect all three channels to be equal. Here you make Y channel the canonical value. Have you tried taking the mean of all three channels? Shouldn't this be a bit better to attenuate the lensfun color shifting?

Reply:
> So in monochrome mode we expect all three channels to be equal

yes.

About taking the mean; i tested that some time ago and there was no visible difference. I looked through the lensfun docs & code and from my understanding the algo makes assumptions from given coeffs how far data are displaced for the different wavelengths of light. So - as green / Y channel is more centric i chose that.

Reply:
Thanks! As a side not it would have been good to have some comments about this, I asked myself this question and I'm sure other devs will do in the future.

Reply:
I'll add some comments.